onnxruntime/core/graph/identity_elimination.cc

```diff
@@ -10,29 +10,15 @@
 namespace onnxruntime {
 
 Status EliminateIdentity::Apply(GraphEditor& graph_editor, Node& node, bool& modified) {
-  std::map<const NodeArg*, NodeArg*> replacement_defs;
-  auto id_input = node.InputDefs()[0];
-  auto id_output = node.OutputDefs()[0];
-  replacement_defs[id_output] = const_cast<NodeArg*>(id_input);
-
-  // Replace (input) defs of the nodes following the Identity with the input to the Identity.
-  for (auto it = node.OutputNodesBegin(), end = node.OutputNodesEnd(); it != end; ++it) {
-    // TODO: Fix the Node API so this operation is supported without resorting to const_cast.
-    const_cast<Node*>(&*it)->ReplaceDefs(replacement_defs);
+  if (graph_editor.RemoveSingleInSingleOutNode(node.Index())) {
     modified = true;
   }
 
-  // Remove the Identity node.
-  graph_editor.RemoveNode(node.Index());
-
-  // TODO: Make sure resolve is not required here.
-  //ONNXRUNTIME_RETURN_IF_ERROR(graph_editor->Resolve());
-
   return Status::OK();
 }
 
-bool EliminateIdentity::SatisfyCondition(const Node& /*node*/) {
-  return true;  // No additional condition required.
+bool EliminateIdentity::SatisfyCondition(const GraphEditor& graph_editor, const Node& node) {
+  return graph_editor.IsSingleInSingleOutNode(node);
```

Comment:
So, does this mean all single/out edge node will be removed please?

Reply:
if yes, looks to me something wrong.

Reply:
As discussed offline, this rule will be applied only to the identity nodes, as this is the node type that this rewrite rule is registered for.